lib/pbench/server/database/models/server_config.py

```diff
@@ -218,12 +211,25 @@ class ServerConfig(Database.Base):
 
     @staticmethod
     def _default(key: str) -> JSONVALUE:
-        if key == ServerConfigOptions.MAXIMUM_DATASET_LIFETIME.key:
-            return str(__class__.config.max_retention_period)
-        elif key == ServerConfigOptions.SERVER_STATE.key:
-            return {STATE_STATUS_KEY: STATE_STATUS_KEYWORD_ENABLED}
-        else:
-            return None
+        try:
+            config = SERVER_CONFIGURATION_OPTIONS[key]
+        except KeyError:
+            if key:
+                raise ServerConfigBadKey(key)
+            else:
+                raise ServerConfigMissingKey()
+        return config["default"]()
+
+    @staticmethod
+    def _validate(key: str, value: JSONVALUE) -> JSONVALUE:
+        try:
+            config = SERVER_CONFIGURATION_OPTIONS[key]
+        except KeyError:
+            if key:
+                raise ServerConfigBadKey(key)
+            else:
+                raise ServerConfigMissingKey()
```

Comment:
No `from`?
```
        except KeyError as k:
            raise ServerConfigBadKey(key) if key else ServerConfigMissingKey() from k
```
Ditto at line 216 and probably elsewhere.

Reply:
I suppose I should get into the habit of froming everywhere, and I worked so hard to get myself out of that habit it doesn't come back easily. 😏